OMMC - 0.6

SkyDynamic opened the pull request:
# Update Summary

## Features 

- Support MC 1.20.6 & 1.21 (By @SkyDynamic)
- Rewrite highlightWaypoint render logic 
- Compat sodium 0.6

## Fixes

- Fix incompatibility with lithium at client side (By @SkyDynamic)
- Fix sortInventory crashes the client due to null tag from data components (By @wendavid552 & @SkyDynamic)
- Fix render issue with sodium

<hr>

# 更新摘要

## 特性

- 支持 MC 1.20.6 & 1.21 (By @SkyDynamic)
- 重写高亮坐标点渲染逻辑
- 兼容 Sodium 0.6

## 修复

- 修复与 lithium 兼容性问题 (By @SkyDynamic)
- 修复整理背包时由于物品组件为空时发生崩溃 (By @wendavid552 & @SkyDynamic)
- 修复世吞挖掘助手部分贴图渲染问题

SkyDynamic committed:
Support MC 1.20.6 & 1.21

SkyDynamic committed:
dump to 0.6

SkyDynamic committed:
fix some problem

SkyDynamic committed:
fix preprocess code

SkyDynamic committed:
fix highlightWaypoint throw "Already building"

SkyDynamic committed:
Make boolean config can bind hotkey

Hendrix-Shen committed:
Rewrite highlightWaypoint render logic

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
Use gradle-all

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
Update buildscripts

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Sunazumi commented:
Game can be broken when using sortInventory
with log here:
[21:27:34] [Render thread/ERROR]: Unreported exception thrown!
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_9279.method_57461()" because "dataA" is null
	at com.plusls.ommc.feature.sortInventory.SortInventoryUtil$ItemStackComparator.compare(SortInventoryUtil.java:430) ~[oh-my-minecraft-client-mc1.21-fabric-0.6.0-dev.jar:?]
	at com.plusls.ommc.feature.sortInventory.SortInventoryUtil$ItemStackComparator.compare(SortInventoryUtil.java:408) ~[oh-my-minecraft-client-mc1.21-fabric-0.6.0-dev.jar:?]
	at java.base/java.util.TimSort.countRunAndMakeAscending(TimSort.java:355) ~[?:?]
	at java.base/java.util.TimSort.sort(TimSort.java:220) ~[?:?]
	at java.base/java.util.Arrays.sort(Arrays.java:1308) ~[?:?]
	at java.base/java.util.ArrayList.sort(ArrayList.java:1804) ~[?:?]
	at com.plusls.ommc.feature.sortInventory.SortInventoryUtil.quickSort(SortInventoryUtil.java:320) ~[oh-my-minecraft-client-mc1.21-fabric-0.6.0-dev.jar:?]
	at com.plusls.ommc.feature.sortInventory.SortInventoryUtil.sort(SortInventoryUtil.java:216) ~[oh-my-minecraft-client-mc1.21-fabric-0.6.0-dev.jar:?]
	at com.plusls.ommc.config.Configs.lambda$init$4(Configs.java:327) ~[oh-my-minecraft-client-mc1.21-fabric-0.6.0-dev.jar:?]
	at top.hendrixshen.magiclib.api.malilib.config.MagicConfigManager.lambda$setHotkeyCallback$0(MagicConfigManager.java:19) ~[magiclib_malilib_extra-0.8.43+31bd62c-beta-11641f01664cbaf2.jar:?]
	at fi.dy.masa.malilib.hotkeys.KeybindMulti.triggerKeyAction(KeybindMulti.java:185) ~[malilib-fabric-1.21-0.19.999-sakura.6.jar:?]
	at fi.dy.masa.malilib.hotkeys.KeybindMulti.updateIsPressed(KeybindMulti.java:145) ~[malilib-fabric-1.21-0.19.999-sakura.6.jar:?]
	at fi.dy.masa.malilib.event.InputEventHandler.checkKeyBindsForChanges(InputEventHandler.java:276) ~[malilib-fabric-1.21-0.19.999-sakura.6.jar:?]
	at fi.dy.masa.malilib.event.InputEventHandler.onKeyInput(InputEventHandler.java:147) ~[malilib-fabric-1.21-0.19.999-sakura.6.jar:?]
	at net.minecraft.class_309.handler$efb000$malilib$onKeyboardInput(class_309.java:1621) ~[client-intermediary.jar:?]
	at net.minecraft.class_309.method_1466(class_309.java:346) ~[client-intermediary.jar:?]
	at net.minecraft.class_309.method_22678(class_309.java:499) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:108) ~[client-intermediary.jar:?]
	at net.minecraft.class_309.method_22676(class_309.java:499) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWKeyCallbackI.callback(GLFWKeyCallbackI.java:44) ~[lwjgl-glfw-3.3.3.jar:?]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.3.jar:?]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3438) ~[lwjgl-glfw-3.3.3.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.pollEvents(RenderSystem.java:150) ~[client-intermediary.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:168) ~[client-intermediary.jar:?]
	at net.minecraft.class_1041.method_15998(class_1041.java:303) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1307) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:882) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:256) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.0.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.0.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.0.jar:?]

Hendrix-Shen committed:
Move ModMenuImpl

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
Fix buildscripts

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen commented:
Due to conflict with #41 and incompatibility with the new version of MC, this PR will be resolved in #41.

Trimera commented:
已收到

FANXING-0710 reviewed (approved):
已收到

Hendrix-Shen committed:
Code style

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Adogs1 commented:
> _未提供任何说明。_

大佬你好，通过你提交的ommc存储库，我成功构建了最新的1.21版本的ommc.jar。我可以以网盘文件的形式对外公布吗？因为大家等ommc的1.21版本太久了。当然，我会标明出处的！

Hendrix-Shen commented:
> > _未提供任何说明。_
> 
> 大佬你好，通过你提交的ommc存储库，我成功构建了最新的1.21版本的ommc.jar。我可以以网盘文件的形式对外公布吗？因为大家等ommc的1.21版本太久了。当然，我会标明出处的！

如果想提前发布请将 `gradle.properties` 中的 `mod.version` 字段修改为 `0.5` 后编译发布

Adogs1 commented:
> > > _未提供任何说明。_
> > 
> > 
> > 大佬你好，通过你提交的ommc存储库，我成功构建了最新的1.21版本的ommc.jar。我可以以网盘文件的形式对外公布吗？因为大家等ommc的1.21版本太久了。当然，我会标明出处的！
> 
> 如果想提前发布请将 中的 字段修改为 后编译发布`gradle.properties``mod.version``0.5`

好的，谢谢

Hendrix-Shen committed:
Disable AccessWideners & interfaceInjection transitive

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
MagicLib 0.8.593+

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
Add missing InitializationHandler

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

lirongrong114514 commented:
大佬你好能不能在项目的Releases里提供构建好的文件我构建一直报网络错误

ghost commented:
大佬你好很多人对构建不了解希望能在项目的Releases里提供构建好的文件方便大家下载谢谢！

xk-mt commented:
> 大你好很多人对构建不了解希望能在项目的发布里提供构建好的文件方便大家下载谢谢！

怎么下载使用，求大佬教一下

ghost commented:
> > 大你好很多人对构建不了解希望能在项目的发布里提供构建好的文件方便大家下载谢谢！
> 
> 怎么下载使用，求大佬教一下

得把仓库源码克隆下载构建对于没学过fabric开发比较复杂我从早上开始构建gradle现在tm还没构建完网速太慢了开代理也没用

lirongrong114514 commented:
> > > 大你好很多人对构建不了解希望能在项目的发布里提供构建好的文件方便大家下载谢谢！
> > 
> > 
> > 怎么下载使用，求大佬教一下
> 
> 得把仓库源码克隆下载构建对于没学过fabric开发比较复杂我从早上开始构建gradle现在tm还没构建完网速太慢了开代理也没用

我甚至一度怀疑是不是卡死了

xk-mt commented:
下载到是有梯子，构建不理解是什么意思，需要我试试吗，我网速还不错

xk-mt commented:
> > > 大你好很多人对构建不了解希望能在项目的发布里提供构建好的文件方便大家下载谢谢！
> > 
> > 
> > 怎么下载使用，求大佬教一下
> 
> 得把仓库源码克隆下载构建对于没学过fabric开发比较复杂我从早上开始构建gradle现在tm还没构建完网速太慢了开代理也没用


下载到是有梯子，构建不理解是什么意思，需要我试试吗，我网速还不错

ghost commented:
构建就是用gradle项目构建器把源码编译成jar软件包也就是mod具体的我也不太懂过程对于没搞过的人比较复杂口头讲不大清楚

xk-mt commented:
> 构建就是用gradle项目构建器把源码编译成jar软件包也就是mod具体的我也不太懂过程对于没搞过的人比较复杂口头讲不大清楚

听不懂，等你编译好分享一份给我，谢谢

Adogs1 commented:
该项目新版本还存在一些问题，我可以提供5天前我编译好的https://wwi.lanzouq.com/iqqjk26t855g，它目前与钠冲突，还有另一个冲突mod我没测出来，这两个mod都可导致闪退，出现情况后欢迎提交issues

xk-mt commented:
> 该项目新版本还存在一些问题，我可以提供5天前我编译好的[https://wwi.lanzouq.com/iqqjk26t855g，它目前与钠冲突，还有另一个冲突mod我没测出来，这两个mod都可导致闪退，出现情况后欢迎提交issues](https://wwi.lanzouq.com/iqqjk26t855g%EF%BC%8C%E5%AE%83%E7%9B%AE%E5%89%8D%E4%B8%8E%E9%92%A0%E5%86%B2%E7%AA%81%EF%BC%8C%E8%BF%98%E6%9C%89%E5%8F%A6%E4%B8%80%E4%B8%AA%E5%86%B2%E7%AA%81mod%E6%88%91%E6%B2%A1%E6%B5%8B%E5%87%BA%E6%9D%A5%EF%BC%8C%E8%BF%99%E4%B8%A4%E4%B8%AAmod%E9%83%BD%E5%8F%AF%E5%AF%BC%E8%87%B4%E9%97%AA%E9%80%80%EF%BC%8C%E5%87%BA%E7%8E%B0%E6%83%85%E5%86%B5%E5%90%8E%E6%AC%A2%E8%BF%8E%E6%8F%90%E4%BA%A4issues)

 - 模组 'Oh My Minecraft Client' (ommc) 0.5.1+7d492db-dev 需要 magiclib 的 0.8.583-beta 及以上版本，但没有安装它！
 
 有一个前置magiclib 模组需要安装，安装后启动世界还是崩了

ghost commented:
> > 该项目新版本还存在一些问题，我可以提供5天前我编译好的[https://wwi.lanzouq.com/iqqjk26t855g，它目前与钠冲突，还有另一个冲突mod我没测出来，这两个mod都可导致闪退，出现情况后欢迎提交issues](https://wwi.lanzouq.com/iqqjk26t855g%EF%BC%8C%E5%AE%83%E7%9B%AE%E5%89%8D%E4%B8%8E%E9%92%A0%E5%86%B2%E7%AA%81%EF%BC%8C%E8%BF%98%E6%9C%89%E5%8F%A6%E4%B8%80%E4%B8%AA%E5%86%B2%E7%AA%81mod%E6%88%91%E6%B2%A1%E6%B5%8B%E5%87%BA%E6%9D%A5%EF%BC%8C%E8%BF%99%E4%B8%A4%E4%B8%AAmod%E9%83%BD%E5%8F%AF%E5%AF%BC%E8%87%B4%E9%97%AA%E9%80%80%EF%BC%8C%E5%87%BA%E7%8E%B0%E6%83%85%E5%86%B5%E5%90%8E%E6%AC%A2%E8%BF%8E%E6%8F%90%E4%BA%A4issues)
> 
> * 模组 'Oh My Minecraft Client' (ommc) 0.5.1+7d492db-dev 需要 magiclib 的 0.8.583-beta 及以上版本，但没有安装它！
> 
> 有一个前置magiclib 模组需要安装，安装后启动世界还是崩了

你装的前置是不是版本太老了我装上去可以正常使用

xk-mt commented:
> > > 该项目新版本还存在一些问题，我可以提供5天前我编译好的[https://wwi.lanzouq.com/iqqjk26t855g，它目前与钠冲突，还有另一个冲突mod我没测出来，这两个mod都可导致闪退，出现情况后欢迎提交issues](https://wwi.lanzouq.com/iqqjk26t855g%EF%BC%8C%E5%AE%83%E7%9B%AE%E5%89%8D%E4%B8%8E%E9%92%A0%E5%86%B2%E7%AA%81%EF%BC%8C%E8%BF%98%E6%9C%89%E5%8F%A6%E4%B8%80%E4%B8%AA%E5%86%B2%E7%AA%81mod%E6%88%91%E6%B2%A1%E6%B5%8B%E5%87%BA%E6%9D%A5%EF%BC%8C%E8%BF%99%E4%B8%A4%E4%B8%AAmod%E9%83%BD%E5%8F%AF%E5%AF%BC%E8%87%B4%E9%97%AA%E9%80%80%EF%BC%8C%E5%87%BA%E7%8E%B0%E6%83%85%E5%86%B5%E5%90%8E%E6%AC%A2%E8%BF%8E%E6%8F%90%E4%BA%A4issues)
> > 
> > 
> > 
> > * 模组 'Oh My Minecraft Client' (ommc) 0.5.1+7d492db-dev 需要 magiclib 的 0.8.583-beta 及以上版本，但没有安装它！
> > 
> > 有一个前置magiclib 模组需要安装，安装后启动世界还是崩了
> 
> 你装的前置是不是版本太老了我装上去可以正常使用

你是java几，游戏什么版本，装了那些mod，方便详细说一下吗，我看看是什么原因

Adogs1 commented:
> > > > 该项目新版本还存在一些问题，我可以提供5天前我编译好的[https://wwi.lanzouq.com/iqqjk26t855g，它目前与钠冲突，还有另一个冲突mod我没测出来，这两个mod都可导致闪退，出现情况后欢迎提交issues](https://wwi.lanzouq.com/iqqjk26t855g%EF%BC%8C%E5%AE%83%E7%9B%AE%E5%89%8D%E4%B8%8E%E9%92%A0%E5%86%B2%E7%AA%81%EF%BC%8C%E8%BF%98%E6%9C%89%E5%8F%A6%E4%B8%80%E4%B8%AA%E5%86%B2%E7%AA%81mod%E6%88%91%E6%B2%A1%E6%B5%8B%E5%87%BA%E6%9D%A5%EF%BC%8C%E8%BF%99%E4%B8%A4%E4%B8%AAmod%E9%83%BD%E5%8F%AF%E5%AF%BC%E8%87%B4%E9%97%AA%E9%80%80%EF%BC%8C%E5%87%BA%E7%8E%B0%E6%83%85%E5%86%B5%E5%90%8E%E6%AC%A2%E8%BF%8E%E6%8F%90%E4%BA%A4issues)
> > > 
> > > 
> > > 
> > > * 模组 'Oh My Minecraft Client' (ommc) 0.5.1+7d492db-dev 需要 magiclib 的 0.8.583-beta 及以上版本，但没有安装它！
> > > 
> > > 有一个前置magiclib 模组需要安装，安装后启动世界还是崩了
> > 
> > 
> > 你装的前置是不是版本太老了我装上去可以正常使用
> 
> 你是java几，游戏什么版本，装了那些mod，方便详细说一下吗，我看看是什么原因

我打包了我可能造成崩溃的全部mod，我再次给你细说造成崩溃流程，其实我在issues都有提到。这是mod下载：我禁用了ommc，以正常游戏，请记得启用它：https://wwow.lanzouq.com/i3UL727d9yri
java版本是zulu21，1.21版本
1.与钠一起使用，会导致进入存档操作，未加载完成后直接崩溃。作者已确认是世吞挖矿助手和模型无偏移的问题
2.不知道与哪个mod一起使用，会导致进入ommc的配置页面（没崩的话随便操作几下，改点东西什么的）后崩溃
以上崩溃表现为客户端直接闪退，报错mod可能冲突

wendavid552 committed:
fix mismatched Operation for @WrapOperation in magiclib

wendavid552 committed:
fix incompatibility with lithium at client side.
Lithium will invoke the getCollisionShape() before the client instant is initialized. This is a hacky fix, preventing the calls to Configs.

wendavid552 commented:
> > > > > 该项目新版本还存在一些问题，我可以提供5天前我编译好的[wwi.lanzouq.com/iqqjk26t855g，它目前与钠冲突，还有另一个冲突mod我没测出来，这两个mod都可导致闪退，出现情况后欢迎提交issues](https://wwi.lanzouq.com/iqqjk26t855g%EF%BC%8C%E5%AE%83%E7%9B%AE%E5%89%8D%E4%B8%8E%E9%92%A0%E5%86%B2%E7%AA%81%EF%BC%8C%E8%BF%98%E6%9C%89%E5%8F%A6%E4%B8%80%E4%B8%AA%E5%86%B2%E7%AA%81mod%E6%88%91%E6%B2%A1%E6%B5%8B%E5%87%BA%E6%9D%A5%EF%BC%8C%E8%BF%99%E4%B8%A4%E4%B8%AAmod%E9%83%BD%E5%8F%AF%E5%AF%BC%E8%87%B4%E9%97%AA%E9%80%80%EF%BC%8C%E5%87%BA%E7%8E%B0%E6%83%85%E5%86%B5%E5%90%8E%E6%AC%A2%E8%BF%8E%E6%8F%90%E4%BA%A4issues)
> > > > 
> > > > 
> > > > 
> > > > * 模组 'Oh My Minecraft Client' (ommc) 0.5.1+7d492db-dev 需要 magiclib 的 0.8.583-beta 及以上版本，但没有安装它！
> > > > 
> > > > 有一个前置magiclib 模组需要安装，安装后启动世界还是崩了
> > > 
> > > 
> > > 你装的前置是不是版本太老了我装上去可以正常使用
> > 
> > 
> > 你是java几，游戏什么版本，装了那些mod，方便详细说一下吗，我看看是什么原因
> 
> 我打包了我可能造成崩溃的全部mod，我再次给你细说造成崩溃流程，其实我在issues都有提到。这是mod下载：我禁用了ommc，以正常游戏，请记得启用它：[wwow.lanzouq.com/i3UL727d9yri](https://wwow.lanzouq.com/i3UL727d9yri) java版本是zulu21，1.21版本 1.与钠一起使用，会导致进入存档操作，未加载完成后直接崩溃。作者已确认是世吞挖矿助手和模型无偏移的问题 2.不知道与哪个mod一起使用，会导致进入ommc的配置页面（没崩的话随便操作几下，改点东西什么的）后崩溃 以上崩溃表现为客户端直接闪退，报错mod可能冲突

以上两个bug已经解决，2其实是和锂一起加载会崩溃，等待pr中

Trimera commented:
已收到

wendavid552 committed:
fix sortInventory crashes the client due to null tag from data components

wendavid552 committed:
update java version for CI build

wendavid552 committed:
fix wrong release files patterns

jittb commented:
> > > 大你好很多人对构建不了解希望能在项目的发布里提供构建好的文件方便大家下载谢谢！
> > 
> > 
> > 怎么下载使用，求大佬教一下
> 
> 得把仓库源码克隆下载构建对于没学过fabric开发比较复杂，我从早上开始构建gradle现在tm还没构建完网速太慢了开代理也没用

有什么文档或者视频能够教学如何构建吗 至少我用idea尝试构建完全是盲人摸象。。

SkyDynamic committed:
Merge pull request #2 from wendavid552/exp/0.6

Fix Nyan-Work#45

SkyDynamic committed:
fix sort data is null

wendavid552 commented:
我根据：[26017dc](https://github.com/Nyan-Work/oh-my-minecraft-client/pull/41/commits/26017dcdeb8228ffc5907d30aceb0c040e86bf55) 通过Github Actions构建了一份，欢迎大家试用，如果有bug的话，请通知到这个pr来，感谢、收到等就不需要发了，谢谢！

I have build one [26017dc](https://github.com/Nyan-Work/oh-my-minecraft-client/pull/41/commits/26017dcdeb8228ffc5907d30aceb0c040e86bf55) from Github Actions, please ping me if any issues.

这是构建的链接，请自取：https://github.com/wendavid552/oh-my-minecraft-client/releases/tag/dev.6

@Lai-pishe  请读一下上面的回答，我已经给了链接了

vocaloidkiss commented:
@wendavid552 由于MagicLib0.8.593的破坏性更新，OMMC目前不能与masagadget共存（僵尸村民导致崩溃）

vocaloidkiss commented:
@wendavid552 请用最新版的修复一下bug吧，或者开放一下issue
目前环境是这样的
[MasaMG] MasaGadget-mc1.21-fabric-4.0.364-stable
[Masa前置] malilib-fabric-1.21.1-0.21.0
[Masa魔法类库] MagicLib-mc1.21.0-fabric-0.8.596-beta

Lai-pishe commented:
有没有大佬构建一个分享一下啊，真的不会搞啊

wendavid552 commented:
> @wendavid552 由于MagicLib0.8.593的破坏性更新，OMMC目前不能与masagadget共存（僵尸村民导致崩溃）

你可以试试看这里的MasaGadget的构建：https://github.com/plusls/MasaGadget/actions/runs/10348149017
具体来讲，就是下载下面的build-artifact，然后在给的压缩包里面找到对应你的mc版本的MasaGadget。
从commit文本来看，你用的公开发布了的MasaGadget版本没支持这个magiclib版本。

Hendrix-Shen committed:
Gradle 8.10.2

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
MC 1.21 -> 1.12.1

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
Fix render issue with sodium

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
Compat sodium 0.6

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
Use public ver of MagicLib

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen commented:
在 c4dc36d 中推送了 Sodium 0.6 的支持，`世吞挖矿助手(worldEaterMineHelper)` 和 `方块模型没有偏移(blockModelNoOffset)` 相关功能应该正常工作

Trimera commented:
已收到

Hendrix-Shen committed:
Clean up

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen committed:
Rewrite CI

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

Hendrix-Shen commented:
目前已经重写 GitHub Actions 构建脚本，本 PR 中的 commit 应当正常触发自动构建，可在 GitHub Actions 对应工作流中下载工件，本更新已经接近尾声，更新摘要已编写到 https://github.com/Nyan-Work/oh-my-minecraft-client/pull/41#issue-2421429938 。如无重大问题本 PR 会于近期合并，后续评论请不要回复诸如“收到”等与问题反馈无关的信息。

注：MC 1.21.3 支持暂不在本 PR 讨论范围之内，先不要再催了哈。

Hendrix-Shen committed:
Fix dev build gh-tag

Signed-off-by: Hendrix-Shen <HendrixShen@hendrixshen.top>

SkyDynamic commented:
目前背包整理无法正常使用 我再尝试修复

Cassy-Lee commented:
> 目前背包整理无法正常使用 我再尝试修复

ItemScroller已有自带物品整理功能，是否可以删除？

Trimera commented:
已收到

Cassy-Lee commented:
是否可以在已有功能时禁用/删除某些功能？
ItemScroller已有物品整理功能
Tweakeroo已有禁止破坏特定方块

Trimera commented:
已收到
